app/src/org/commcare/engine/cases/AndroidCaseInstanceTreeElement.java

```diff
@@ -91,8 +95,9 @@ protected synchronized void loadElements() {
         String[] namesToMatch = new String[numKeys];
         String[] valuesToMatch = new String[numKeys];
         for (int i = numKeys - 1; i >= 0; i--) {
-            namesToMatch[i] = keys.elementAt(i);
-            valuesToMatch[i] = (String)values.elementAt(i);
+            namesToMatch[i] = profiles.elementAt(i).getKey();
+            valuesToMatch[i] = (String)

```

Comment:
Hm, this is a good question, this is pretty sketchy, actually, although I'm a bit hesitant to change the old behavior right now. I'm going to add this to a round 2 backlog for revisions that I'm going to make to make all of this indexing stuff less crazy.